src/lib/BLE/devBLETelemetry.cpp

```diff
@@ -0,0 +1,301 @@
+#if defined(PLATFORM_ESP32)
+
+#include "CRSF.h"
+#include "FHSS.h"
+#include "NimBLEDevice.h"
+#include "POWERMGNT.h"
+#include "common.h"
+#include "hwTimer.h"
+#include "logging.h"
+#include "options.h"
+
+#include "devBLETelemetry.h"
+
+#include "config.h"
+extern TxConfig config;
+
+NimBLEServer *pServer;
+NimBLECharacteristic *rcCRSF;
+
+#define DURATION_UPDATE 100  // timeout after 100ms
+
+//how often link stats packet is sent over BLE
+#define BLE_LINKSTATS_PACKET_PERIOD_MS 500
+
+//how often channels(sticks positions) packet is sent over BLE
+#define BLE_CHANNELS_PACKET_PERIOD_MS 300
+
+unsigned short const TELEMETRY_SVC_UUID = 0x1819;
+unsigned short const TELEMETRY_CRSF_UUID = 0x2BBD;
+
+unsigned short const DEVICE_INFO_SVC_UUID = 0x180A;
+unsigned short const MODEL_NUMBER_SVC_UUID = 0x2A24;
+unsigned short const SERIAL_NUMBER_SVC_UUID = 0x2A25;
+unsigned short const SOFTWARE_NUMBER_SVC_UUID = 0x2A28;
+unsigned short const HARDWARE_NUMBER_SVC_UUID = 0x2A27;
+unsigned short const MANUFACTURER_NAME_SVC_UUID = 0x2A29;
+
+
+static uint32_t LastTMLLinkStatsPacketMillis = 0;
+static uint32_t LastTLMRCPacketMillis = 0;
+static bool initOnce = false;
+
+static SemaphoreHandle_t _mutex = nullptr;
```

Comment:
```suggestion
static SemaphoreHandle_t _mutex = nullptr;

class BLEServerCB : public NimBLEServerCallbacks {
  void onConnect(NimBLEServer *pServer, ble_gap_conn_desc  *param) override 
  {
    if ( pServer != nullptr)
    {
        NimBLEDevice::startAdvertising();
    }
  }
} bleServerCB;
```

Continue advertising on connection. This allows connecting multiple devices, .f.e using tablet + ground station + antena tracker.

Reply:
added